db-models/comments.sql

```diff
@@ -1,9 +1,12 @@
 -- db-models/comments.sql
 
 CREATE TABLE comments (
-    comment_id SERIAL PRIMARY KEY,          -- Unique for each comment
-    parent_comment_id INTEGER,              -- ID of the parent comment for replies
-    author_name TEXT NOT NULL,              -- Author comment's name
-    content TEXT NOT NULL,                  -- Comment text
-    created_at TIMESTAMP DEFAULT NOW()      --Timestamp when comment was created
+    comment_id INTEGER PRIMARY KEY AUTOINCREMENT,                   -- Unique for each comment, auto-incremented
+    parent_comment_id INTEGER,                                      -- ID of the parent comment for replies
+    parent_listing_id INTEGER NOT NULL,                             -- ID of the parent comment listing, required
+    creator_id INTEGER NOT NULL,                                    -- ID of the author, required
+    content TEXT NOT NULL,                                          -- Comment text
+    created_at TEXT DEFAULT CURRENT_TIMESTAMP NOT NULL,             -- Timestamp when comment was created
+    FOREIGN KEY (creator_id) REFERENCES creators(id),
```

Comment:
The `creators` table you're referencing here should likely be `users`. And in the users table, the ID column will be `user_id` like you did in the line below this one